src/tribler/core/components/metadata_store/remote_query_community/settings.py

```diff
@@ -5,7 +5,7 @@ class RemoteQueryCommunitySettings(TriblerConfigSection):
     minimal_blob_size: int = 200
     maximum_payload_size: int = 1300
     max_entries: int = maximum_payload_size // minimal_blob_size
-    max_query_peers: int = 5
+    max_query_peers: int = 20
```

Comment:
I'd like to give a simple answer to this question, but unfortunately, it is a bit complicated.

`max_query_peers` is the number of peers to which `GigaChannelCommunity` (inherited from `RemoteQueryCommunity`) sends remote search requests. Currently, only the `GigaChannelCommunity` subclass of `RemoteQueryCommunity` uses this option ([here](https://github.com/Tribler/tribler/blob/dd8a622f8b8f8be18e0bad24c430b16326e1d0f1/src/tribler/core/components/gigachannel/community/gigachannel_community.py#L223)); the other subclass is `PopularityCommunity` that does not use the option. 

Requests are sent and aggregated entirely in parallel, so as long as on average there are not too many concurrent requests coming to the same peer, there is no reason to limit the number of peers to which we send requests. On average, it is better to send requests to all connected peers to have a variety of results in responses.

Currently, 20 peers is the target number of remote peers for the GigaChannelCommunity. This is because GigaChannelCommunity [uses the RemovePeers strategy](https://github.com/Tribler/tribler/blob/3a49a9ec42396c52e57532ffd5d239e8e94a910c/src/tribler/core/components/gigachannel/gigachannel_component.py#L44), and it [aims to reduce the number of peers to 20](https://github.com/Tribler/tribler/blob/9fe7cf8e3478b9142f2be40c076ce98088021f8e/src/tribler/core/components/gigachannel/community/sync_strategy.py#L16). In PR, I introduce the named constant `TARGET_PEERS_NUMBER` for that number.

At the time of launch, GigaChannelCommunity uses [DiscoverBooster](https://github.com/Tribler/tribler/blob/dd8a622f8b8f8be18e0bad24c430b16326e1d0f1/src/tribler/core/components/gigachannel/community/gigachannel_community.py#L110), thanks to which the number of `GigaChannelCommunity` peers in the first minutes after the launch of Tribler can reach several hundred. I don't think sending parallel search requests to a few hundred peers is necessary a good idea, so I decided to limit the maximum number of concurrent requests, and the number of `TARGET_PEERS_NUMBER` peers looks like a good option.

It is possible to import the `TARGET_PEERS_NUMBER` constant to use it as a value for `max_query_peers` option in `RemoteQueryCommunitySettings`, but it also requires some additional considerations, as we do not import other settings values in a similar way. For now, I added a comment to the `max_query_peers` option that refers to `TARGET_PEERS_NUMBER`.

Now a few words about moving `max_query_peers` to a more appropriate place in settings.

As only `GigaChannelCommunity` uses this option, it is possible to move it from the `RemoteQueryCommunity` settings to the [settings](https://github.com/Tribler/tribler/blob/dd8a622f8b8f8be18e0bad24c430b16326e1d0f1/src/tribler/core/components/gigachannel/community/settings.py#L7) of `GigaChannelCommunity`, but it provokes additional questions and for that reason in my opinion lies outside of the scope of the current PR. Two examples of questions:

1. should we have `RemoteQueryCommunitySettings` at all, as it is essentially an abstract class, and settings should probably be specified in a concrete subclass, to be able to have different settings in different concrete community subclasses (right now, `GigaChannelCommunity` and `PopularityCommunity`)
2. should we put search-related settings in a separate config section, as the search functionality is not essentially tied to `GigaChannelCommunity`?

In my opinion, the option initially was placed in the wrong config section, but moving it to a correct section is a task of another possible PR.

Reply:
Thank you for the explanation 🙏 now `20` is not a magic number anymore.